include/boost/gil/extension/numeric/algorithm.hpp

```diff
@@ -248,6 +250,143 @@ void view_multiplies_scalar(SrcView const& src_view, Scalar const& scalar, DstVi
     }
 }
 
+
+/// \ingroup ImageAlgorithms
+/// \brief Boundary options for image boundary extension
+enum class boundary_option
+{
+    boundary_option_extend_zero,    /// assume the source boundaries to be zero
+    boundary_option_extend_constant /// assume the source boundaries to be the boundary value
+};
+
+namespace detail
+{
+
+template <typename SrcView, typename RltView>
+void extend_row_impl(
+    SrcView src_view,
+    RltView result_view,
+    std::size_t extend_count,
+    boundary_option option)
+{
+    if (option == boundary_option::boundary_option_extend_constant)
+    {
+        for (std::size_t i = 0; i < static_cast<std::size_t>(result_view.height()); i++)
+        {
+            if(i >= extend_count && i < extend_count + src_view.height())
+            {
+                assign_pixels(
+                    src_view.row_begin(i - extend_count), 
```

Comment:
All indexes within image view dimensions (`coord_t`, `x_coord_t`, `y_coord_t`, etc.) are (historically) derived in GIL from `std::ptrdiff_t`, not `std::size_t`. So, the use of `std::size_t` may lead to annoying compilation warnings.

Please, ensure this new code does not add up to the already existing warnings flood.

Reply:
I actually had used `std::ptrdiff_t` diff earlier but it gave me warning due to comparison with `extend_count` which is of type `std::size_t` so looped using `std::size_t` and converted height and width explicitly. 

If you insist on using `std::ptrdiff_t` then I will have to convert `extend_count` to `std::ptrdiff_t` from `std::size_t` to avoid warning.

Reply:
>If you insist on using std::ptrdiff_t then I will have to convert extend_count to std::ptrdiff_t from std::size_t to avoid warning.

I insist on having no warnings. If it means (safe) casting or changing types, your choice.
If you prefer `size_t extend_count` , then you can cast internally to `ptrdiff_t`.


In detail, my point is:
Using common types for indexing pixel rows/columns should be preferred.
Currently, the common type is derived from `std::ptrdiff_t`. It may change in future - there were discussions where @stefanseefeld suggested better approach or use of `std::size_t` - but _before that happens, any new code should stick to current conventions_.

Reply:
Finally, I converted `extend_count` to `std::ptrdiff_t` and also checked there are no warning related to it.

Reply:
@lpranam  Thank you!